### PR TITLE
fix: [Document] sort rules are not respected during filter by name and extended search - EXO-62937.

### PR DESCRIPTION
**Pull request by akhanfir**
Prior to this change,the filter by name in the list of document files and extended search is not working properly. After this changes, this list of document files displays the workflows correctly.

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
The pr have conflict, can you update ?

Can you also rework the commit message ?
In your message, you have this 

> Prior to this change,the filter by name in the list of document files and extended search is not working properly

It does not explain WHY the filter not working. Please try to answer question WHY

and this :

> After this changes, this list of document files displays the workflows correctly.

It does not explain HOW you fix the problem. Please try to reword by answering the question HOW.

Finally, you try to fix the problem by sorting the search result list, after the search. 
But, in documentSearchServiceConnector, we pass options sortField and sortDirection. So the result list should be sorted by the search connector. 

In addition, my IDE show this 
image

The sortField and sortDirection are not used in the search function. I think it could be better to use it in the ES query, so that ES is in charge of sorting the result, and we get already sorted results, no need to recompute it.

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you rebase the pr

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/search/DocumentSearchServiceConnector.java [hunk 1]
```diff
@@ -117,7 +117,11 @@ public Collection<SearchResult> search(Identity userIdentity,
     if (StringUtils.isBlank(filter.getQuery()) && !filter.getFavorites()) {
       throw new IllegalArgumentException("Filter term is mandatory");
     }
-    String esQuery = buildQueryStatement(userIdentity, workspace, path, filter, offset, limit);
+    if (!sortField.equalsIgnoreCase("lastUpdatedDate")) {
+      sortField="_score";
+    }
```
Comment (rdenarie) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/search/DocumentSearchServiceConnector.java:129]: if sortField is title, you must sort on title.raw

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**